restli-common/src/main/java/com/linkedin/restli/internal/common/URIParamUtils.java

```diff
@@ -62,7 +62,14 @@
     {
       if (dataMap.containsKey(parameterName))
       {
-        result.put(parameterName, URIMaskUtil.encodeMaskForURI(dataMap.getDataMap(parameterName)));
+        try
```

Comment:
Using try-catches to dictate normal logic flow (rather than just for exceptions) is bad practice. I would suggest retrieving the generic object from the map and checking its type.

Overall though this is better IMO.

Reply:
Thanks for the advice, fixed it.